### PR TITLE
Generate a date for tests in ForecastFilterTest instead of using a fixed one

### PR DESCRIPTION
**Pull request by daniellimws**
Generate the current day's date for tasks that are tested for filtering by week and by day. Also did some refactoring to reduce repeated code and renamed some methods to make them more descriptive.

I think ideally we should test that 7 days in the week (Monday to Sunday) are included when filtering by week. I haven't added this yet as I just wanted the CI to pass for now.

### CODE REVIEW

**Comment by daniellimws** [comment]
Fix #98

File: src/test/java/athena/task/taskfilter/ForecastFilterTest.java [hunk 1]
```diff
@@ -24,37 +30,39 @@ void testIsTaskIncluded_all_returnsTrue() {
...
-     * Checks if task is not included after applying the week forecast filter.
+     * Checks if task not in this week is not included after applying the week forecast filter.
      */
     @Test
-    void testIsTaskIncluded_week_returnsFalse() {
+    void testIsTaskIncluded_filterByWeek_returnsFalse() {
         ForecastFilter forecastFilter = new ForecastFilter(Forecast.WEEK);
         Task inputTask = new Task("testName", "0900", "1", "05-11-2020",
-                "23-10-2020", Importance.LOW, "testNotes", 0); // Tested on 12-10-2020
+                "23-10-2020", Importance.LOW, "testNotes", 0);
```
Comment (wish2023) [inline src/test/java/athena/task/taskfilter/ForecastFilterTest.java:52]: This test will return true once 23-10-2020 is in the same week as the current date. Perhaps there's a better way to generate a false case. Maybe the recurrence can be `(LocalDate.now() + 7 days).toString()`

File: src/test/java/athena/task/taskfilter/ForecastFilterTest.java [hunk 2]
```diff
@@ -24,37 +30,39 @@ void testIsTaskIncluded_all_returnsTrue() {
...
+     * Checks if task in this week is included after applying the week forecast filter.
      */
     @Test
-    void testIsTaskIncluded_week_returnsTrue() {
+    void testIsTaskIncluded_filterByWeek_returnsTrue() {
+        String todayDateString = LocalDate.now().toString();
         ForecastFilter forecastFilter = new ForecastFilter(Forecast.WEEK);
         Task inputTask = new Task("testName", "0900", "1", "05-11-2020",
-                "16-10-2020", Importance.LOW, "testNotes", 0); // Tested on 12-10-2020
+                todayDateString, Importance.LOW, "testNotes", 0);
```
Comment (wish2023) [inline src/test/java/athena/task/taskfilter/ForecastFilterTest.java:40]: This test is exactly the same as `testIsTaskIncluded_filterByToday_returnsTrue()`. Perhaps a better input could be a day that lies in the same week as `LocalDate.now` and adding this as a parameter.

File: src/test/java/athena/tasklist/TaskListTest.java [hunk 3]
```diff
@@ -51,119 +53,125 @@ void editTask_givenAttributes_attributeChanged() throws TaskNotFoundException {
...
         } else if (importance == Importance.MEDIUM) {
-            taskList.addTask(new Task("dos", "1100",
-                    "2 hours", "16-09-2020", "15-10-2020", Importance.MEDIUM,
-                    "Refer to slides", 1));
+            taskList.addTask(task2);
         } else if (importance == Importance.LOW) {
-            taskList.addTask(new Task("tres", "1100",
-                    "2 hours", "16-09-2020", "13-11-2020", Importance.LOW,
-                    "Refer to slides", 2));
+            taskList.addTask(task3);
```
Comment (wish2023) [inline src/test/java/athena/tasklist/TaskListTest.java:132]: Good idea to simplify the code

File: text-ui-test/EXPECTED.TXT [hunk 4]
```diff
@@ -40,16 +40,16 @@ Additionally, you've also added these additional notes!
 Refer to lecture notes.
 Looks like another mission to complete! Let's do it!
 
-2020-10-13:
+2020-10-17:
```
Comment (wish2023) [inline text-ui-test/EXPECTED.TXT:43]: Will we have to change the dates EXPECTED.TXT everytime we test?

**Review by wish2023 — CHANGES_REQUESTED** [review]
Minor changes have been suggested

File: src/test/java/athena/task/taskfilter/ForecastFilterTest.java (hunk 1 above)
Comment (daniellimws) [reply src/test/java/athena/task/taskfilter/ForecastFilterTest.java:52]: Oh right good catch

File: text-ui-test/EXPECTED.TXT (hunk 4 above)
Comment (daniellimws) [reply text-ui-test/EXPECTED.TXT:43]: I removed the runtest.sh from GitHub actions in 732c687 because I realized this is not practical.

**Comment by daniellimws** [comment]
@wish2023 I'm thinking of merging this first so that we can move on with the other PRs. Will open another issue to address the comments you left. Ok?

**Pull request merged**